### ISSUE TITLE
old gradient fill shows if chart is scrolled offscreen

### ISSUE THREAD

**Issue by dam13n** [Jawbone/JBChartView#204]
I have a line chart that I can toggle between showing different sets of data, and it's in a scrollview. If the chart view is scrolled offscreen and then becomes visible again, the last gradient fill(s) remain and a new one is drawn on top. This does not affect the redrawing of lines though.

**Comment by terryworona** [comment]
I sort of understand, but before I try to reproduce, you mind sending detailed steps to reproduce? Expected and actual results?

Demo code or project would also help.

**Comment by dam13n** [comment]
Demo code is gonna be tough because I'm using RubyMotion and I have a rather complex setup of JBChartView. Three different charts w/ one flipped. I'll start w/ showing a quick video and can provide further explanation and some code if needed. : )

BTW, JBChartView is such a sweet library.

https://dl.dropboxusercontent.com/u/599100/gradient_fill_bug.m4v

**Comment by terryworona** [comment]
Hrm. Got it. 

The chart's are largely static, unless you call reloadData/reloadDataAnimated:

Meaning, they don't redraw or anything if they go off screen. Layout is calculated at runtime of reloadData based on the existing frame and bounds. 

Are you calling reloadData when you scroll back? Not saying that's wrong -- just staying it's a good place to start debugging.

**Comment by dam13n** [comment]
I only call reloadData when I am hitting those buttons ( overall / business / personal) and when initializing them.

Anyway, I was totally off track on what caused the issue. It wasn't the scrolling offscreen, it was this in the `- (void)touchesEndedOrCancelledWithTouches:(NSSet *)touches` method:
```
[self.linesView setSelectedLineIndex:kJBLineChartLinesViewUnselectedLineIndex animated:YES];
```
So just panning reproduces the error me. Sorry about that.

To get three charts working together, I had to disable user interaction on them, and manually call `touch` methods using gesture recognizers. By setting `showsLineSelection` to NO, I've resolved this issue. I'm curious to look into `setSelectedLineIndex` further, but I haven't yet.
